specs/SharedBuffer.md

```diff
@@ -0,0 +1,282 @@
+  ///
+  /// Sharing a buffer to script has security risk. You should only share buffer with trusted site.
+  /// 
+  HRESULT ShareBufferToScript(
+    [in] ICoreWebView2StagingSharedBuffer* sharedBuffer,
+    [in] BOOL isReadOnlyToScript,
+    [in] LPCWSTR additionalDataAsJson);
+}
+
+interface ICoreWebView2StagingFrame2 : IUnknown {
+  /// Share a shared buffer object with script of the iframe in the WebView.
+  /// The script will receive a `SharedBufferReceived` event from chrome.webview.
+  /// The event arg for that event will have the following properties:
+  ///   `sharedBuffer`: an ArrayBuffer object with the backing content from the shared buffer.
+  ///   `data`: an object representing `additionalDataAsJson`. This property will be `undefined`
+  ///           if `additionalDataAsJson` is nullptr or empty string.
+  ///   `source`: with a value set as `chrome.webview` object.
+  /// If `isReadOnlyToScript` is true, the script will only have read access to the buffer.
+  /// If the script tries to modify the content in a read only buffer, it will cause an access violation
+  /// in WebView renderer process and crash the renderer process.
+  /// If the shared buffer is already closed, the API will fail with HRESULT_FROM_WIN32(ERROR_INVALID_STATE).
+  /// 
+  /// The script code should call `chrome.webview.releaseSharedBuffer` with
+  /// the shared buffer as the parameter to release underlying resources as soon
+  /// as it does not need access the shared buffer any more.
+  ///
+  /// Sharing a buffer to script has security risk. You should only share buffer with trusted site.
+  /// 
+  HRESULT ShareBufferToScript(
+    [in] ICoreWebView2StagingSharedBuffer* sharedBuffer,
+    [in] BOOL isReadOnlyToScript,
+    [in] LPCWSTR additionalDataAsJson);
+}
+
+```
+
+## .NET
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+
+    class CoreWebView2Environment
+    {
+        public CoreWebView2SharedBuffer CreateSharedBuffer(ulong Size);
+    }
+
+    class CoreWebView2SharedBuffer
+    {
+        public ulong Size { get; };
+        public IntPtr Buffer { get; };
+        public IntPtr Handle { get; };
+        public Stream GetAccessStream();
+        void Close();
+    }
+    
+    runtimeclass CoreWebView2
+    {
+        public void ShareBufferToScript(CoreWebView2SharedBuffer sharedBuffer, bool isReadOnlyToScript, string additionalDataAsJson);
+    }
+    
+    class CoreWebView2Frame
+    {
+        public void ShareBufferToScript(CoreWebView2SharedBuffer sharedBuffer, bool isReadOnlyToScript, string additionalDataAsJson);
+    }
+}
+
+```
+## WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+
+    runtimeclass CoreWebView2Environment
+    {
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2StagingEnvironment2")]
+        {
+            CoreWebView2SharedBuffer CreateSharedBuffer(UInt64 Size);
+        }
+    }
+
+    runtimeclass CoreWebView2SharedBuffer
```

Comment:
It should probably implement Windows.Foundation.IClosable

And for the .NET one it should probably implement IDisposable

Reply:
Make sense, will add those.